test/blas.jl

```diff
@@ -3,35 +3,36 @@
 import Base.LinAlg, Base.LinAlg.BlasReal, Base.LinAlg.BlasComplex
 
 srand(100)
-# syr2k! and her2k!
-for elty in (Float32, Float64, Complex64, Complex128)
-    U = randn(5,2)
-    V = randn(5,2)
-    if elty == Complex64 || elty == Complex128
-        U = complex.(U, U)
-        V = complex.(V, V)
+@testset "syr2k! and her2k!" begin
```

Comment:
I think we'd get more informative errors if you keep the separation of `syr2k` and `her2k`, pull the `for` loop into the `@testset` (instead of using `begin`), and interpolate `$elty` into the string.